recipe/magento2.php

```diff
@@ -69,7 +76,39 @@
 
 desc('Upgrade magento database');
 task('magento:upgrade:db', function () {
-    run("{{bin/php}} {{release_path}}/bin/magento setup:upgrade --keep-generated");
+    $databaseUpgradeNeeded = false;
+
+    try {
+        run('{{bin/php}} {{release_path}}/bin/magento setup:db:status');
+    } catch (ProcessFailedException $e) {
+        if ($e->getProcess()->getExitCode() == DB_UPDATE_NEEDED_EXIT_CODE) {
+            $databaseUpgradeNeeded = true;
+        } else {
+            throw $e;
+        }
+    } catch (RunException $e) {
+        if ($e->getExitCode() == DB_UPDATE_NEEDED_EXIT_CODE) {
+            $databaseUpgradeNeeded = true;
+        } else {
+            throw $e;
+        }
+    }
+
+    if ($databaseUpgradeNeeded) {
+        // check whether maintenance mode is enabled or not in order to keep the status afterwards
+        $maintenanceModeStatusOutput = run("{{bin/php}} {{release_path}}/bin/magento maintenance:status");
```

Comment:
Wouldn't it be quicker to check for the maintenance flag instead of kicking off another bin/magento command?

Reply:
Indeed it would be, but I decided to stick to the framework tools.  Do you see any other cons beside speed?

Reply:
Not really, this is fine, just wanted to know the reason :)